### ISSUE TITLE
Accessibility quirks in various versions of Eset NOD32 with NVDA

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#963]
**Reported by Palacee_hun on 2010-10-04 21:07**
System: Hungarian Windows XP SP3 32 bit
NVDA: 2010.2 branch, portable snapshot 3867
However I think these do not matter too much for this ticket.

To experience the problems, you can do the following:
1. Download a free 30-day trial version of NOD32 Antivirus (I did this from [http://www.eset.com.au/download/trial_versions.html], this is an English language version, its version number is 4.2.64.12)
2. Install the software. This should pose no problems since the installer seems quite accessible.
3. After finishing the installation, the NOD32 UI starts up automatically in a window calledEset NOD32 Antivirus Application. NVDA announces the title correctly.
4. According to the docs, this UI has two versions: a graphical (this is the default) and a more accessible text one. You can switch between these by hitting Ctrl-G. So hit it to enter the "more accessible" text mode.
5. Try to interact with the UI by cursor keys, tabbing, reviewing and using flat review.
With sighted assistance you can see that the UI reacts to cursoring and tabbing, but NVDA remains totally silent. It should announce the item currently selected from the list on the left part of the window.
Reviewing only reports the window title, trying to invoke flat review fails and tells me that this is not available on this object.
1. Try to move the mouse over the UI window. This makes NVDA to announce some tooltips for the items under the mouse pointer, but this behaviour is not reliable. NVDA even reads some other static text like parts of the window also on a seemingly random basis.

It is hard to reproduce the problems in connection with the warning windows of NOD32 due to the nature of them. When I browsed some webpages in Firefox, the HTTP monitor component of NOD32 detected something suspicious and popped up a window asking me what to do next (e.g. whether I permit it to submit the suspicious data to Eset for analysis). However I only deduce this from what NVDA has read about the warning window (only some words). After I tried to tab through the warning window, Windows displayed a crash window telling me that EGUI (the UI program) had crashed and had been stopped. Launching the UI from the Start Menu was ossible.

Some notes:
It seems that NVDA does not find the elements of the NOD32 UI window.
Despite this UI inaccessibility, NOD32 runs fine and updates its virus database smoothly. Fortunately I have not had the opportunity to try out what NOD32 would do if confronted with a real threat, would the inaccessibility of the threat alert window stop it from preventing the activation of the threat.
I have found that it does not matter whether NOD32 UI is in graphical or text mode, the results are the same, it is virtually inaccessible either way.

I suggest a "major" severity degree for this ticket, as it is too well known that antimalware software is essential for Windows, and NOD32 is among the best according to multiple test forums and it contains not only antivirus, but antispyware as well.
Oddly enough, Eset claims on their website that the latest 4.2 version had UI improvements to ensure smooth cooperation with screen readers.

**Comment by leonardder** [comment]
@k-kolev1985: Do you agree with closing as works for me? I'm afraid we can't track down the original author.

**Comment by k-kolev1985** [comment]
@leonardder Well, here we are talking about older and probably not supported by ESET versions of NOD32. We may as well close it. I don't use NOD32 anymore, due to several reasons:
1. I've changed my internet provider and the current one does not provide me with a free license for NOD32, as did the previous one.
2. Last I checked, the latest versions of NOD32 are not accessible to use with NVDA.
I leave it to you to decide what to do with this ticket.

**Comment by leonardder** [comment]
Thanks, closing for now. Please open a new issue for accessibility quirks in recent versions of Nod32.
